x/mongo/driver/compression.go

```diff
@@ -123,7 +130,13 @@ func CompressPayload(in []byte, opts CompressionOpts) ([]byte, error) {
 		if err != nil {
 			return nil, err
 		}
-		return encoder.EncodeAll(in, nil), nil
+		ptr := zstdBufPool.Get().(*[]byte)
+		b := encoder.EncodeAll(in, *ptr)
```

Comment:
Instead of adding an additional pool of buffers just for Zstd compression, can we leverage the [existing op buffer pool](https://github.com/mongodb/mongo-go-driver/blob/27aacc59cbd0016ccf6e40dbda698f543d4ff1c7/x/mongo/driver/operation.go#L740) to improve memory usage for _all_ compression types?

For example, in `compression.go`, change `CompressPayload` to accept a destination byte slice:
```go
func CompressPayload(src, dst []byte, opts CompressionOpts) ([]byte, error) {
	// ...
	return encoder.EncodeAll(src, dst)
	// ...
```
Then in `connection.go`, pass the existing destination byte slice into `CompressPayload`:
```go
func (c *Connection) CompressWireMessage(src, dst []byte) ([]byte, error) {
	// ...
	compressed, err := driver.CompressPayload(rem, dst, opts)
	// ...
```
Then remove the following call to `AppendCompressedCompressedMessage`, which is just an alias for `append`.

Reply:
The benchmark shows no improvements after passing the `dst` slice to `CompressPayload`. I also tried replacing `zstdBufPool` with the existing `memoryPool`, or applying `zstdBufPool` on other compression types, but none improved memory consumption.